// vue-notion/dist/ssr.js
'use strict';Object.defineProperty(exports,'__esModule',{value:true});var Vue=require('vue'),Prism=require('prismjs'),PrismComponent=require('vue-prism-component'),vueFragment=require('vue-fragment'),fetch=require('cross-fetch');function _interopDefaultLegacy(e){return e&&typeof e==='object'&&'default'in e?e:{'default':e}}var Vue__default=/*#__PURE__*/_interopDefaultLegacy(Vue);var Prism__default=/*#__PURE__*/_interopDefaultLegacy(Prism);var PrismComponent__default=/*#__PURE__*/_interopDefaultLegacy(PrismComponent);var fetch__default=/*#__PURE__*/_interopDefaultLegacy(fetch);function ownKeys(object, enumerableOnly) {
  var keys = Object.keys(object);

  if (Object.getOwnPropertySymbols) {
    var symbols = Object.getOwnPropertySymbols(object);

    if (enumerableOnly) {
      symbols = symbols.filter(function (sym) {
        return Object.getOwnPropertyDescriptor(object, sym).enumerable;
      });
    }

    keys.push.apply(keys, symbols);
  }

  return keys;
}

function _objectSpread2(target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i] != null ? arguments[i] : {};

    if (i % 2) {
      ownKeys(Object(source), true).forEach(function (key) {
        _defineProperty(target, key, source[key]);
      });
    } else if (Object.getOwnPropertyDescriptors) {
      Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
      ownKeys(Object(source)).forEach(function (key) {
        Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
      });
    }
  }

  return target;
}

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) {
  try {
    var info = gen[key](arg);
    var value = info.value;
  } catch (error) {
    reject(error);
    return;
  }

  if (info.done) {
    resolve(value);
  } else {
    Promise.resolve(value).then(_next, _throw);
  }
}

function _asyncToGenerator(fn) {
  return function () {
    var self = this,
        args = arguments;
    return new Promise(function (resolve, reject) {
      var gen = fn.apply(self, args);

      function _next(value) {
        asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value);
      }

      function _throw(err) {
        asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err);
      }

      _next(undefined);
    });
  };
}

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};

  var target = _objectWithoutPropertiesLoose(source, excluded);

  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

function _slicedToArray(arr, i) {
  return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest();
}

function _arrayWithHoles(arr) {
  if (Array.isArray(arr)) return arr;
}

function _iterableToArrayLimit(arr, i) {
  var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"];

  if (_i == null) return;
  var _arr = [];
  var _n = true;
  var _d = false;

  var _s, _e;

  try {
    for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) {
      _arr.push(_s.value);

      if (i && _arr.length === i) break;
    }
  } catch (err) {
    _d = true;
    _e = err;
  } finally {
    try {
      if (!_n && _i["return"] != null) _i["return"]();
    } finally {
      if (_d) throw _e;
    }
  }

  return _arr;
}

function _unsupportedIterableToArray(o, minLen) {
  if (!o) return;
  if (typeof o === "string") return _arrayLikeToArray(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor) n = o.constructor.name;
  if (n === "Map" || n === "Set") return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen);
}

function _arrayLikeToArray(arr, len) {
  if (len == null || len > arr.length) len = arr.length;

  for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i];

  return arr2;
}

function _nonIterableRest() {
  throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}// utils from react-notion
var getTextContent = function getTextContent(text) {
  return text.reduce(function (prev, current) {
    return prev + current[0];
  }, "");
};

var groupBlockContent = function groupBlockContent(blockMap) {
  var output = [];
  var lastType = undefined;
  var index = -1;
  Object.keys(blockMap).forEach(function (id) {
    var _blockMap$id$value$co;

    (_blockMap$id$value$co = blockMap[id].value.content) === null || _blockMap$id$value$co === void 0 ? void 0 : _blockMap$id$value$co.forEach(function (blockId) {
      var _blockMap$blockId, _blockMap$blockId$val;

      var blockType = (_blockMap$blockId = blockMap[blockId]) === null || _blockMap$blockId === void 0 ? void 0 : (_blockMap$blockId$val = _blockMap$blockId.value) === null || _blockMap$blockId$val === void 0 ? void 0 : _blockMap$blockId$val.type;

      if (blockType && blockType !== lastType) {
        index++;
        lastType = blockType;
        output[index] = [];
      }

      output[index].push(blockId);
    });
    lastType = undefined;
  });
  return output;
};

var getListNumber = function getListNumber(blockId, blockMap) {
  var groups = groupBlockContent(blockMap);
  var group = groups.find(function (g) {
    return g.includes(blockId);
  });

  if (!group) {
    return;
  }

  return group.indexOf(blockId) + 1;
};
var defaultMapImageUrl = function defaultMapImageUrl() {
  var image = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : "";
  var block = arguments.length > 1 ? arguments[1] : undefined;
  var url = new URL("https://www.notion.so".concat(image.startsWith("/image") ? image : "/image/".concat(encodeURIComponent(image))));

  if (block && !image.includes("/images/page-cover/")) {
    var table = block.value.parent_table === "space" ? "block" : block.value.parent_table;
    url.searchParams.set("table", table);
    url.searchParams.set("id", block.value.id);
    url.searchParams.set("cache", "v2");
  }

  return url.toString();
};
var defaultMapPageUrl = function defaultMapPageUrl() {
  var pageId = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : "";
  pageId = pageId.replace(/-/g, "");
  return "/".concat(pageId);
};var blockProps = {
  blockMap: {
    type: Object,
    required: true
  },
  blockOverrides: {
    type: Object,
    default: function _default() {
      return {};
    }
  },
  contentId: {
    type: String,
    required: false
  },
  contentIndex: {
    type: Number,
    default: 0
  },
  embedAllow: {
    type: String,
    default: "fullscreen"
  },
  fullPage: {
    type: Boolean,
    default: false
  },
  hideList: {
    type: Array,
    default: function _default() {
      return [];
    }
  },
  imageOptions: Object,
  katex: {
    type: Boolean,
    default: false
  },
  level: {
    type: Number,
    default: 0
  },
  mapImageUrl: Function,
  mapPageUrl: Function,
  pageLinkOptions: Object,
  pageLinkTarget: {
    type: String,
    default: "_self"
  },
  prism: {
    type: Boolean,
    default: false
  },
  textLinkTarget: {
    type: String,
    default: "_blank"
  },
  todo: {
    type: Boolean,
    default: false
  }
};
var blockComputed = {
  pass: function pass() {
    // todo: make this more dynamic by iterating over blockProps attributes
    return {
      blockMap: this.blockMap,
      blockOverrides: this.blockOverrides,
      contentId: this.contentId,
      contentIndex: this.contentIndex,
      embedAllow: this.embedAllow,
      fullPage: this.fullPage,
      hideList: this.hideList,
      imageOptions: this.imageOptions,
      katex: this.katex,
      level: this.level,
      mapImageUrl: this.mapImageUrl,
      mapPageUrl: this.mapPageUrl,
      pageLinkOptions: this.pageLinkOptions,
      prism: this.prism,
      todo: this.todo
    };
  },
  alt: function alt() {
    var _this$caption;

    return (_this$caption = this.caption) === null || _this$caption === void 0 ? void 0 : _this$caption[0][0];
  },
  block: function block() {
    var id = this.contentId || Object.keys(this.blockMap)[0];
    return this.blockMap[id];
  },
  value: function value() {
    var _this$block;

    return (_this$block = this.block) === null || _this$block === void 0 ? void 0 : _this$block.value;
  },
  format: function format() {
    var _this$value;

    return (_this$value = this.value) === null || _this$value === void 0 ? void 0 : _this$value.format;
  },
  f: function f() {
    var _this$format, _this$format2, _this$format3, _this$format4, _this$format5, _this$format6, _this$format7;

    // format with defaults if empty
    return {
      block_aspect_ratio: (_this$format = this.format) === null || _this$format === void 0 ? void 0 : _this$format.block_aspect_ratio,
      block_height: ((_this$format2 = this.format) === null || _this$format2 === void 0 ? void 0 : _this$format2.block_height) || 1,
      block_width: ((_this$format3 = this.format) === null || _this$format3 === void 0 ? void 0 : _this$format3.block_width) || 1,
      block_color: (_this$format4 = this.format) === null || _this$format4 === void 0 ? void 0 : _this$format4.block_color,
      bookmark_icon: (_this$format5 = this.format) === null || _this$format5 === void 0 ? void 0 : _this$format5.bookmark_icon,
      bookmark_cover: (_this$format6 = this.format) === null || _this$format6 === void 0 ? void 0 : _this$format6.bookmark_cover,
      display_source: (_this$format7 = this.format) === null || _this$format7 === void 0 ? void 0 : _this$format7.display_source
    };
  },
  icon: function icon() {
    var _this$format8;

    return ((_this$format8 = this.format) === null || _this$format8 === void 0 ? void 0 : _this$format8.page_icon) || "";
  },
  width: function width() {
    var _this$format9;

    return (_this$format9 = this.format) === null || _this$format9 === void 0 ? void 0 : _this$format9.block_width;
  },
  properties: function properties() {
    var _this$value2;

    return (_this$value2 = this.value) === null || _this$value2 === void 0 ? void 0 : _this$value2.properties;
  },
  caption: function caption() {
    var _this$properties;

    return (_this$properties = this.properties) === null || _this$properties === void 0 ? void 0 : _this$properties.caption;
  },
  description: function description() {
    var _this$properties2;

    return (_this$properties2 = this.properties) === null || _this$properties2 === void 0 ? void 0 : _this$properties2.description;
  },
  src: function src() {
    var _this$properties3;

    return this.mapImageUrl((_this$properties3 = this.properties) === null || _this$properties3 === void 0 ? void 0 : _this$properties3.source[0][0], this.block);
  },
  title: function title() {
    var _this$properties4;

    return (_this$properties4 = this.properties) === null || _this$properties4 === void 0 ? void 0 : _this$properties4.title;
  },
  type: function type() {
    var _this$value3;

    return (_this$value3 = this.value) === null || _this$value3 === void 0 ? void 0 : _this$value3.type;
  },
  visible: function visible() {
    return !this.hideList.includes(this.type);
  },
  hasPageLinkOptions: function hasPageLinkOptions() {
    var _this$pageLinkOptions, _this$pageLinkOptions2;

    return ((_this$pageLinkOptions = this.pageLinkOptions) === null || _this$pageLinkOptions === void 0 ? void 0 : _this$pageLinkOptions.component) && ((_this$pageLinkOptions2 = this.pageLinkOptions) === null || _this$pageLinkOptions2 === void 0 ? void 0 : _this$pageLinkOptions2.href);
  },
  parent: function parent() {
    var _this$value4;

    return this.blockMap[(_this$value4 = this.value) === null || _this$value4 === void 0 ? void 0 : _this$value4.parent_id];
  }
};
var Blockable = {
  props: blockProps,
  computed: blockComputed,
  methods: {
    getTextContent: getTextContent,
    isType: function isType(t) {
      if (Array.isArray(t)) {
        return this.visible && t.includes(this.type);
      }

      return this.visible && this.type === t;
    },
    blockColorClass: function blockColorClass() {
      var _this$format10;

      var suffix = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : "";
      var blockColor = (_this$format10 = this.format) === null || _this$format10 === void 0 ? void 0 : _this$format10.block_color;
      return blockColor ? "notion-".concat(blockColor).concat(suffix) : undefined;
    },
    pageLinkProps: function pageLinkProps(id) {
      var _this$pageLinkOptions3;

      return _defineProperty({}, ((_this$pageLinkOptions3 = this.pageLinkOptions) === null || _this$pageLinkOptions3 === void 0 ? void 0 : _this$pageLinkOptions3.href) || "href", this.mapPageUrl(id));
    }
  }
};var script$q = {
  extends: Blockable,
  name: "NotionDecorator",
  props: _objectSpread2(_objectSpread2({}, blockProps), {}, {
    content: Array
  }),
  computed: {
    text: function text() {
      var _this$content;

      return (_this$content = this.content) === null || _this$content === void 0 ? void 0 : _this$content[0];
    },
    decorators: function decorators() {
      var _this$content2;

      return ((_this$content2 = this.content) === null || _this$content2 === void 0 ? void 0 : _this$content2[1]) || [];
    },
    decoratorKey: function decoratorKey() {
      var _this$decorators, _this$decorators$;

      return (_this$decorators = this.decorators) === null || _this$decorators === void 0 ? void 0 : (_this$decorators$ = _this$decorators[0]) === null || _this$decorators$ === void 0 ? void 0 : _this$decorators$[0];
    },
    decoratorValue: function decoratorValue() {
      var _this$decorators2, _this$decorators2$;

      return (_this$decorators2 = this.decorators) === null || _this$decorators2 === void 0 ? void 0 : (_this$decorators2$ = _this$decorators2[0]) === null || _this$decorators2$ === void 0 ? void 0 : _this$decorators2$[1];
    },
    unappliedDecorators: function unappliedDecorators() {
      var clonedDecorators = JSON.parse(JSON.stringify(this.decorators || []));
      clonedDecorators.shift(); // remove applied decorator

      return clonedDecorators;
    },
    nextContent: function nextContent() {
      return [this.text, this.unappliedDecorators];
    },
    isPageLink: function isPageLink() {
      return this.text === "‣";
    },
    isInlinePageLink: function isInlinePageLink() {
      return this.decoratorValue && this.decoratorValue[0] === "/";
    },
    pageLinkTitle: function pageLinkTitle() {
      var _this$blockMap, _this$blockMap$this$d, _this$blockMap$this$d2, _this$blockMap$this$d3, _this$blockMap$this$d4, _this$blockMap$this$d5;

      return ((_this$blockMap = this.blockMap) === null || _this$blockMap === void 0 ? void 0 : (_this$blockMap$this$d = _this$blockMap[this.decoratorValue]) === null || _this$blockMap$this$d === void 0 ? void 0 : (_this$blockMap$this$d2 = _this$blockMap$this$d.value) === null || _this$blockMap$this$d2 === void 0 ? void 0 : (_this$blockMap$this$d3 = _this$blockMap$this$d2.properties) === null || _this$blockMap$this$d3 === void 0 ? void 0 : (_this$blockMap$this$d4 = _this$blockMap$this$d3.title) === null || _this$blockMap$this$d4 === void 0 ? void 0 : (_this$blockMap$this$d5 = _this$blockMap$this$d4[0]) === null || _this$blockMap$this$d5 === void 0 ? void 0 : _this$blockMap$this$d5[0]) || "link";
    },
    target: function target() {
      if (this.type === "page") {
        return this.pageLinkTarget;
      }

      return this.textLinkTarget;
    }
  }
};function normalizeComponent(template, style, script, scopeId, isFunctionalTemplate, moduleIdentifier /* server only */, shadowMode, createInjector, createInjectorSSR, createInjectorShadow) {
    if (typeof shadowMode !== 'boolean') {
        createInjectorSSR = createInjector;
        createInjector = shadowMode;
        shadowMode = false;
    }
    // Vue.extend constructor export interop.
    const options = typeof script === 'function' ? script.options : script;
    // render functions
    if (template && template.render) {
        options.render = template.render;
        options.staticRenderFns = template.staticRenderFns;
        options._compiled = true;
        // functional template
        if (isFunctionalTemplate) {
            options.functional = true;
        }
    }
    // scopedId
    if (scopeId) {
        options._scopeId = scopeId;
    }
    let hook;
    if (moduleIdentifier) {
        // server build
        hook = function (context) {
            // 2.3 injection
            context =
                context || // cached call
                    (this.$vnode && this.$vnode.ssrContext) || // stateful
                    (this.parent && this.parent.$vnode && this.parent.$vnode.ssrContext); // functional
            // 2.2 with runInNewContext: true
            if (!context && typeof __VUE_SSR_CONTEXT__ !== 'undefined') {
                context = __VUE_SSR_CONTEXT__;
            }
            // inject component styles
            if (style) {
                style.call(this, createInjectorSSR(context));
            }
            // register component module identifier for async chunk inference
            if (context && context._registeredComponents) {
                context._registeredComponents.add(moduleIdentifier);
            }
        };
        // used by ssr in case component is cached and beforeCreate
        // never gets called
        options._ssrRegister = hook;
    }
    else if (style) {
        hook = shadowMode
            ? function (context) {
                style.call(this, createInjectorShadow(context, this.$root.$options.shadowRoot));
            }
            : function (context) {
                style.call(this, createInjector(context));
            };
    }
    if (hook) {
        if (options.functional) {
            // register for functional component in vue file
            const originalRender = options.render;
            options.render = function renderWithStyleInjection(h, context) {
                hook.call(context);
                return originalRender(h, context);
            };
        }
        else {
            // inject component registration as beforeCreate hook
            const existing = options.beforeCreate;
            options.beforeCreate = existing ? [].concat(existing, hook) : [hook];
        }
    }
    return script;
}/* script */
var __vue_script__$q = script$q;
/* template */

var __vue_render__$p = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _vm.isPageLink && _vm.hasPageLinkOptions ? _c(_vm.pageLinkOptions.component, _vm._b({
    tag: "component",
    staticClass: "notion-link"
  }, 'component', _vm.pageLinkProps(_vm.decoratorValue), false), [_vm._v("\n  " + _vm._s(_vm.pageLinkTitle) + "\n")]) : _vm.isPageLink ? _c('a', {
    staticClass: "notion-link",
    attrs: {
      "target": _vm.pageLinkTarget,
      "href": _vm.mapPageUrl(_vm.decoratorValue)
    }
  }, [_vm._v(_vm._s(_vm.pageLinkTitle))]) : _vm.decoratorKey === 'a' && _vm.hasPageLinkOptions && _vm.isInlinePageLink ? _c(_vm.pageLinkOptions.component, _vm._b({
    tag: "component",
    staticClass: "notion-link"
  }, 'component', _vm.pageLinkProps(_vm.decoratorValue.slice(1)), false), [_c('NotionDecorator', _vm._b({
    attrs: {
      "content": _vm.nextContent
    }
  }, 'NotionDecorator', _vm.pass, false))], 1) : _vm.decoratorKey === 'a' && _vm.isInlinePageLink ? _c('a', {
    staticClass: "notion-link",
    attrs: {
      "target": _vm.target,
      "href": _vm.mapPageUrl(_vm.decoratorValue.slice(1))
    }
  }, [_c('NotionDecorator', _vm._b({
    attrs: {
      "content": _vm.nextContent
    }
  }, 'NotionDecorator', _vm.pass, false))], 1) : _vm.decoratorKey === 'a' ? _c('a', {
    staticClass: "notion-link",
    attrs: {
      "target": _vm.target,
      "href": _vm.decoratorValue
    }
  }, [_c('NotionDecorator', _vm._b({
    attrs: {
      "content": _vm.nextContent
    }
  }, 'NotionDecorator', _vm.pass, false))], 1) : _vm.decorators.length === 0 ? _c('span', [_vm._v(_vm._s(_vm.text))]) : _vm.decoratorKey === 'h' ? _c('span', {
    class: 'notion-' + _vm.decoratorValue
  }, [_c('NotionDecorator', _vm._b({
    attrs: {
      "content": _vm.nextContent
    }
  }, 'NotionDecorator', _vm.pass, false))], 1) : _vm.decoratorKey === 'c' ? _c('code', {
    staticClass: "notion-inline-code"
  }, [_c('NotionDecorator', _vm._b({
    attrs: {
      "content": _vm.nextContent
    }
  }, 'NotionDecorator', _vm.pass, false))], 1) : _vm.decoratorKey === 'b' ? _c('b', [_c('NotionDecorator', _vm._b({
    attrs: {
      "content": _vm.nextContent
    }
  }, 'NotionDecorator', _vm.pass, false))], 1) : _vm.decoratorKey === 'i' ? _c('em', [_c('NotionDecorator', _vm._b({
    attrs: {
      "content": _vm.nextContent
    }
  }, 'NotionDecorator', _vm.pass, false))], 1) : _vm.decoratorKey === 's' ? _c('s', [_c('NotionDecorator', _vm._b({
    attrs: {
      "content": _vm.nextContent
    }
  }, 'NotionDecorator', _vm.pass, false))], 1) : _vm.decoratorKey === 'e' && _vm.katex ? _c('katex-element', {
    tag: "component",
    attrs: {
      "expression": _vm.decoratorValue
    }
  }) : _vm.decoratorKey === 'e' ? _c('code', {
    staticClass: "notion-inline-code"
  }, [_vm._v("\n  " + _vm._s(_vm.decoratorValue) + "\n")]) : _c('NotionDecorator', _vm._b({
    attrs: {
      "content": _vm.nextContent
    }
  }, 'NotionDecorator', _vm.pass, false));
};

var __vue_staticRenderFns__$p = [];
/* style */

var __vue_inject_styles__$q = undefined;
/* scoped */

var __vue_scope_id__$q = undefined;
/* module identifier */

var __vue_module_identifier__$q = "data-v-3a088875";
/* functional template */

var __vue_is_functional_template__$q = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$q = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$p,
  staticRenderFns: __vue_staticRenderFns__$p
}, __vue_inject_styles__$q, __vue_script__$q, __vue_scope_id__$q, __vue_is_functional_template__$q, __vue_module_identifier__$q, false, undefined, undefined, undefined);var script$p = {
  extends: Blockable,
  name: "NotionTextRenderer",
  props: _objectSpread2(_objectSpread2({}, blockProps), {}, {
    text: Array
  }),
  components: {
    NotionDecorator: __vue_component__$q
  }
};/* script */
var __vue_script__$p = script$p;
/* template */

var __vue_render__$o = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('span', _vm._l(_vm.text, function (t, i) {
    return _c('NotionDecorator', _vm._b({
      key: i,
      attrs: {
        "content": t
      }
    }, 'NotionDecorator', _vm.pass, false));
  }), 1);
};

var __vue_staticRenderFns__$o = [];
/* style */

var __vue_inject_styles__$p = undefined;
/* scoped */

var __vue_scope_id__$p = undefined;
/* module identifier */

var __vue_module_identifier__$p = "data-v-434e9bc9";
/* functional template */

var __vue_is_functional_template__$p = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$p = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$o,
  staticRenderFns: __vue_staticRenderFns__$o
}, __vue_inject_styles__$p, __vue_script__$p, __vue_scope_id__$p, __vue_is_functional_template__$p, __vue_module_identifier__$p, false, undefined, undefined, undefined);//
var script$o = {
  extends: Blockable,
  name: "NotionBookmark",
  components: {
    NotionTextRenderer: __vue_component__$p
  }
};/* script */
var __vue_script__$o = script$o;
/* template */

var __vue_render__$n = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('div', {
    staticClass: "notion-row"
  }, [_vm._ssrNode("<a target=\"_blank\" rel=\"noopener noreferrer\"" + _vm._ssrAttr("href", _vm.properties.link) + _vm._ssrClass(null, ['notion-bookmark', _vm.f.block_color && "notion-" + _vm.f.block_color]) + ">", "</a>", [_vm._ssrNode("<div>", "</div>", [_vm._ssrNode("<div class=\"notion-bookmark-title\">", "</div>", [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title || _vm.properties.link
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1), _vm._ssrNode(" "), _vm.description ? _vm._ssrNode("<div class=\"notion-bookmark-description\">", "</div>", [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.description
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1) : _vm._e(), _vm._ssrNode(" "), _vm._ssrNode("<div class=\"notion-bookmark-link\">", "</div>", [_vm._ssrNode((_vm.f.bookmark_icon ? "<img" + _vm._ssrAttr("alt", _vm.getTextContent(_vm.title || _vm.properties.link)) + _vm._ssrAttr("src", _vm.f.bookmark_icon) + ">" : "<!---->") + " "), _vm._ssrNode("<div>", "</div>", [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.properties.link
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1)], 2)], 2), _vm._ssrNode(" " + (_vm.f.bookmark_cover ? "<div class=\"notion-bookmark-image\"><img" + _vm._ssrAttr("alt", _vm.getTextContent(_vm.title || _vm.properties.link)) + _vm._ssrAttr("src", _vm.f.bookmark_cover) + "></div>" : "<!---->"))], 2)]);
};

var __vue_staticRenderFns__$n = [];
/* style */

var __vue_inject_styles__$o = undefined;
/* scoped */

var __vue_scope_id__$o = undefined;
/* module identifier */

var __vue_module_identifier__$o = "data-v-3e7d278a";
/* functional template */

var __vue_is_functional_template__$o = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$o = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$n,
  staticRenderFns: __vue_staticRenderFns__$n
}, __vue_inject_styles__$o, __vue_script__$o, __vue_scope_id__$o, __vue_is_functional_template__$o, __vue_module_identifier__$o, false, undefined, undefined, undefined);var script$n = {
  extends: Blockable,
  name: "NotionPageIcon",
  props: _objectSpread2(_objectSpread2({}, blockProps), {}, {
    big: Boolean
  })
};/* script */
var __vue_script__$n = script$n;
/* template */

var __vue_render__$m = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _vm.icon.includes('http') ? _c('img', {
    class: [_vm.format.page_cover && 'notion-page-icon-offset', _vm.big ? 'notion-page-icon-cover' : 'notion-page-icon'],
    attrs: {
      "src": _vm.mapImageUrl(_vm.icon, _vm.block),
      "alt": _vm.title ? _vm.getTextContent(_vm.title) : 'Icon'
    }
  }) : _vm.icon ? _c('span', {
    class: ['notion-emoji', _vm.format.page_cover && 'notion-page-icon-offset', _vm.big ? 'notion-page-icon-cover' : 'notion-page-icon'],
    attrs: {
      "role": "img",
      "aria-label": _vm.icon
    }
  }, [_vm._ssrNode(_vm._ssrEscape("\n  " + _vm._s(_vm.icon) + "\n"))]) : _vm._e();
};

var __vue_staticRenderFns__$m = [];
/* style */

var __vue_inject_styles__$n = undefined;
/* scoped */

var __vue_scope_id__$n = undefined;
/* module identifier */

var __vue_module_identifier__$n = "data-v-1f607a53";
/* functional template */

var __vue_is_functional_template__$n = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$n = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$m,
  staticRenderFns: __vue_staticRenderFns__$m
}, __vue_inject_styles__$n, __vue_script__$n, __vue_scope_id__$n, __vue_is_functional_template__$n, __vue_module_identifier__$n, false, undefined, undefined, undefined);//
var script$m = {
  extends: Blockable,
  name: "NotionCallout",
  components: {
    NotionPageIcon: __vue_component__$n,
    NotionTextRenderer: __vue_component__$p
  }
};/* script */
var __vue_script__$m = script$m;
/* template */

var __vue_render__$l = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('div', {
    class: ['notion-callout', _vm.blockColorClass(), _vm.blockColorClass('_co')]
  }, [_vm._ssrNode("<div>", "</div>", [_c('NotionPageIcon', _vm._b({}, 'NotionPageIcon', _vm.pass, false))], 1), _vm._ssrNode(" "), _vm._ssrNode("<div class=\"notion-callout-text\">", "</div>", [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1)], 2);
};

var __vue_staticRenderFns__$l = [];
/* style */

var __vue_inject_styles__$m = undefined;
/* scoped */

var __vue_scope_id__$m = undefined;
/* module identifier */

var __vue_module_identifier__$m = "data-v-790f9fde";
/* functional template */

var __vue_is_functional_template__$m = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$m = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$l,
  staticRenderFns: __vue_staticRenderFns__$l
}, __vue_inject_styles__$m, __vue_script__$m, __vue_scope_id__$m, __vue_is_functional_template__$m, __vue_module_identifier__$m, false, undefined, undefined, undefined);var script$l = {
  extends: Blockable,
  name: "NotionCode",
  components: {
    PrismComponent: PrismComponent__default['default']
  },
  props: _objectSpread2(_objectSpread2({}, blockProps), {}, {
    overrideLang: String,
    overrideLangClass: String
  }),
  data: function data() {
    return {
      Prism: Prism__default['default']
    };
  },
  computed: _objectSpread2(_objectSpread2({}, blockComputed), {}, {
    lang: function lang() {
      var _this$properties, _this$properties$lang, _this$properties$lang2, _this$properties$lang3;

      return this.overrideLang || ((_this$properties = this.properties) === null || _this$properties === void 0 ? void 0 : (_this$properties$lang = _this$properties.language) === null || _this$properties$lang === void 0 ? void 0 : (_this$properties$lang2 = _this$properties$lang[0]) === null || _this$properties$lang2 === void 0 ? void 0 : (_this$properties$lang3 = _this$properties$lang2[0]) === null || _this$properties$lang3 === void 0 ? void 0 : _this$properties$lang3.toLowerCase());
    },
    langClass: function langClass() {
      return this.overrideLangClass || "language-".concat(this.lang);
    },
    supported: function supported() {
      return this.Prism.languages[this.lang];
    }
  })
};/* script */
var __vue_script__$l = script$l;
/* template */

var __vue_render__$k = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _vm.prism && _vm.supported ? _c('pre', {
    class: ['notion-code', _vm.langClass]
  }, [_c('PrismComponent', {
    attrs: {
      "language": _vm.lang
    }
  }, [_vm._v(_vm._s(_vm.properties.title[0][0]))])], 1) : _c('pre', {
    class: ['notion-code', _vm.langClass]
  }, [_vm._ssrNode("<code" + _vm._ssrClass(null, _vm.langClass) + ">" + _vm._ssrEscape(_vm._s(_vm.properties.title[0][0])) + "</code>")]);
};

var __vue_staticRenderFns__$k = [];
/* style */

var __vue_inject_styles__$l = undefined;
/* scoped */

var __vue_scope_id__$l = undefined;
/* module identifier */

var __vue_module_identifier__$l = "data-v-decbf5ac";
/* functional template */

var __vue_is_functional_template__$l = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$l = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$k,
  staticRenderFns: __vue_staticRenderFns__$k
}, __vue_inject_styles__$l, __vue_script__$l, __vue_scope_id__$l, __vue_is_functional_template__$l, __vue_module_identifier__$l, false, undefined, undefined, undefined);//
var baseWidth$1 = 46; // todo: add magic numbers to a config json

var script$k = {
  name: "NotionColumn",
  props: ["format"],
  components: {
    Fragment: vueFragment.Fragment
  },
  computed: {
    columnStyle: function columnStyle() {
      var columns = Number((1 / this.format.column_ratio).toFixed(0));
      var totalWidth = (columns - 1) * baseWidth$1;
      return {
        width: "calc((100% - ".concat(totalWidth, "px) * ").concat(this.format.column_ratio, ")")
      };
    },
    spacerStyle: function spacerStyle() {
      return {
        width: "".concat(baseWidth$1, "px")
      };
    }
  }
};/* script */
var __vue_script__$k = script$k;
/* template */

var __vue_render__$j = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('Fragment', [_c('div', {
    staticClass: "notion-column",
    style: _vm.columnStyle
  }, [_vm._t("default")], 2), _vm._v(" "), _c('div', {
    staticClass: "notion-spacer",
    style: _vm.spacerStyle
  })]);
};

var __vue_staticRenderFns__$j = [];
/* style */

var __vue_inject_styles__$k = undefined;
/* scoped */

var __vue_scope_id__$k = undefined;
/* module identifier */

var __vue_module_identifier__$k = "data-v-311c46b0";
/* functional template */

var __vue_is_functional_template__$k = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$k = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$j,
  staticRenderFns: __vue_staticRenderFns__$j
}, __vue_inject_styles__$k, __vue_script__$k, __vue_scope_id__$k, __vue_is_functional_template__$k, __vue_module_identifier__$k, false, undefined, undefined, undefined);var script$j = {
  extends: Blockable,
  name: "NotionEquation",
  components: {
    NotionCode: __vue_component__$l
  },
  computed: _objectSpread2(_objectSpread2({}, blockComputed), {}, {
    equation: function equation() {
      var _this$properties, _this$properties$titl, _this$properties$titl2;

      return (_this$properties = this.properties) === null || _this$properties === void 0 ? void 0 : (_this$properties$titl = _this$properties.title) === null || _this$properties$titl === void 0 ? void 0 : (_this$properties$titl2 = _this$properties$titl[0]) === null || _this$properties$titl2 === void 0 ? void 0 : _this$properties$titl2[0];
    }
  })
};/* script */
var __vue_script__$j = script$j;
/* template */

var __vue_render__$i = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _vm.katex ? _c('div', [_c('katex-element', {
    tag: "component",
    attrs: {
      "expression": _vm.equation
    }
  })], 1) : _c('NotionCode', _vm._b({
    attrs: {
      "overrideLang": "latex",
      "overrideLangClass": "language-latex"
    }
  }, 'NotionCode', _vm.pass, false));
};

var __vue_staticRenderFns__$i = [];
/* style */

var __vue_inject_styles__$j = undefined;
/* scoped */

var __vue_scope_id__$j = undefined;
/* module identifier */

var __vue_module_identifier__$j = "data-v-20db0f82";
/* functional template */

var __vue_is_functional_template__$j = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$j = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$i,
  staticRenderFns: __vue_staticRenderFns__$i
}, __vue_inject_styles__$j, __vue_script__$j, __vue_scope_id__$j, __vue_is_functional_template__$j, __vue_module_identifier__$j, false, undefined, undefined, undefined);var script$i = {
  extends: Blockable,
  name: "NotionAsset",
  computed: _objectSpread2(_objectSpread2({}, blockComputed), {}, {
    src: function src() {
      return this.type === "figma" ? this.properties.source[0][0] : this.f.display_source;
    },
    style: function style() {
      var aspectRatio = this.f.block_aspect_ratio || this.f.block_height / this.f.block_width;
      return {
        paddingBottom: "".concat(aspectRatio * 100, "%"),
        position: "relative"
      };
    }
  })
};/* script */
var __vue_script__$i = script$i;
/* template */

var __vue_render__$h = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('div', {
    style: _vm.style
  }, [_vm._ssrNode("<iframe" + _vm._ssrAttr("src", _vm.src) + _vm._ssrAttr("allow", _vm.embedAllow) + " class=\"notion-image-inset\"></iframe>")]);
};

var __vue_staticRenderFns__$h = [];
/* style */

var __vue_inject_styles__$i = undefined;
/* scoped */

var __vue_scope_id__$i = undefined;
/* module identifier */

var __vue_module_identifier__$i = "data-v-5a7f8352";
/* functional template */

var __vue_is_functional_template__$i = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$i = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$h,
  staticRenderFns: __vue_staticRenderFns__$h
}, __vue_inject_styles__$i, __vue_script__$i, __vue_scope_id__$i, __vue_is_functional_template__$i, __vue_module_identifier__$i, false, undefined, undefined, undefined);var _excluded = ["component"];
var script$h = {
  extends: Blockable,
  name: "NotionImage",
  computed: _objectSpread2(_objectSpread2({}, blockComputed), {}, {
    hasImageComponent: function hasImageComponent() {
      var _this$imageOptions;

      return !!((_this$imageOptions = this.imageOptions) !== null && _this$imageOptions !== void 0 && _this$imageOptions.component);
    },
    imageProps: function imageProps() {
      var _this$imageOptions2;

      var _ref = this.imageOptions || {};
          _ref.component;
          var attrs = _objectWithoutProperties(_ref, _excluded);

      return _objectSpread2(_objectSpread2({}, attrs), {}, _defineProperty({}, ((_this$imageOptions2 = this.imageOptions) === null || _this$imageOptions2 === void 0 ? void 0 : _this$imageOptions2.src) || "src", this.src));
    },
    style: function style() {
      var aspectRatio = this.f.block_aspect_ratio || this.f.block_height / this.f.block_width;
      return {
        paddingBottom: "".concat(aspectRatio * 100, "%"),
        position: "relative"
      };
    }
  })
};/* script */
var __vue_script__$h = script$h;
/* template */

var __vue_render__$g = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _vm.f.block_aspect_ratio ? _c('div', {
    style: _vm.style
  }, [_vm.hasImageComponent ? _c(_vm.imageOptions.component, _vm._b({
    tag: "component",
    staticClass: "notion-image-inset",
    attrs: {
      "alt": _vm.alt || 'Notion image'
    }
  }, 'component', _vm.imageProps, false)) : _c('img', _vm._b({
    staticClass: "notion-image-inset",
    attrs: {
      "alt": _vm.alt || 'Notion image'
    }
  }, 'img', _vm.imageProps, false))], 1) : _vm.hasImageComponent ? _c(_vm.imageOptions.component, _vm._b({
    tag: "component",
    attrs: {
      "alt": _vm.alt || 'Notion image'
    }
  }, 'component', _vm.imageProps, false)) : _c('img', _vm._b({
    attrs: {
      "alt": _vm.alt
    }
  }, 'img', _vm.imageProps, false), []);
};

var __vue_staticRenderFns__$g = [];
/* style */

var __vue_inject_styles__$h = undefined;
/* scoped */

var __vue_scope_id__$h = undefined;
/* module identifier */

var __vue_module_identifier__$h = "data-v-6ec305e5";
/* functional template */

var __vue_is_functional_template__$h = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$h = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$g,
  staticRenderFns: __vue_staticRenderFns__$g
}, __vue_inject_styles__$h, __vue_script__$h, __vue_scope_id__$h, __vue_is_functional_template__$h, __vue_module_identifier__$h, false, undefined, undefined, undefined);//
var script$g = {
  extends: Blockable,
  name: "NotionFigure",
  components: {
    NotionAsset: __vue_component__$i,
    NotionImage: __vue_component__$h,
    NotionTextRenderer: __vue_component__$p
  }
};/* script */
var __vue_script__$g = script$g;
/* template */

var __vue_render__$f = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('figure', {
    staticClass: "notion-asset-wrapper",
    style: _vm.width
  }, [_vm.isType('image') ? _c('NotionImage', _vm._b({}, 'NotionImage', _vm.pass, false)) : _vm.isType(['embed', 'video', 'figma']) ? _c('NotionAsset', _vm._b({}, 'NotionAsset', _vm.pass, false)) : _vm._e(), _vm._ssrNode(" "), _vm.caption ? _vm._ssrNode("<figcaption class=\"notion-image-caption\">", "</figcaption>", [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.caption
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1) : _vm._e()], 2);
};

var __vue_staticRenderFns__$f = [];
/* style */

var __vue_inject_styles__$g = undefined;
/* scoped */

var __vue_scope_id__$g = undefined;
/* module identifier */

var __vue_module_identifier__$g = "data-v-66c35f72";
/* functional template */

var __vue_is_functional_template__$g = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$g = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$f,
  staticRenderFns: __vue_staticRenderFns__$f
}, __vue_inject_styles__$g, __vue_script__$g, __vue_scope_id__$g, __vue_is_functional_template__$g, __vue_module_identifier__$g, false, undefined, undefined, undefined);//
var script$f = {
  extends: Blockable,
  name: "NotionHeader",
  components: {
    NotionTextRenderer: __vue_component__$p
  }
};/* script */
var __vue_script__$f = script$f;
/* template */

var __vue_render__$e = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _vm.type === 'header' ? _c('h1', {
    staticClass: "notion-h1"
  }, [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1) : _vm.type === 'sub_header' ? _c('h2', {
    staticClass: "notion-h2"
  }, [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1) : _vm.type === 'sub_sub_header' ? _c('h3', {
    staticClass: "notion-h3"
  }, [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1) : _vm._e();
};

var __vue_staticRenderFns__$e = [];
/* style */

var __vue_inject_styles__$f = undefined;
/* scoped */

var __vue_scope_id__$f = undefined;
/* module identifier */

var __vue_module_identifier__$f = "data-v-b46bf420";
/* functional template */

var __vue_is_functional_template__$f = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$f = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$e,
  staticRenderFns: __vue_staticRenderFns__$e
}, __vue_inject_styles__$f, __vue_script__$f, __vue_scope_id__$f, __vue_is_functional_template__$f, __vue_module_identifier__$f, false, undefined, undefined, undefined);var script$e = {
  extends: Blockable,
  name: "NotionNestedList",
  computed: _objectSpread2(_objectSpread2({}, blockComputed), {}, {
    start: function start() {
      var _this$value;

      return getListNumber((_this$value = this.value) === null || _this$value === void 0 ? void 0 : _this$value.id, this.blockMap);
    }
  })
};/* script */
var __vue_script__$e = script$e;
/* template */

var __vue_render__$d = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _vm.type === 'bulleted_list' ? _c('ul', {
    staticClass: "notion-list notion-list-disc"
  }, [_vm._t("default")], 2) : _c('ol', {
    staticClass: "notion-list notion-list-numbered"
  }, [_vm._t("default")], 2);
};

var __vue_staticRenderFns__$d = [];
/* style */

var __vue_inject_styles__$e = undefined;
/* scoped */

var __vue_scope_id__$e = undefined;
/* module identifier */

var __vue_module_identifier__$e = "data-v-7df6e103";
/* functional template */

var __vue_is_functional_template__$e = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$e = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$d,
  staticRenderFns: __vue_staticRenderFns__$d
}, __vue_inject_styles__$e, __vue_script__$e, __vue_scope_id__$e, __vue_is_functional_template__$e, __vue_module_identifier__$e, false, undefined, undefined, undefined);var script$d = {
  extends: Blockable,
  name: "NotionList",
  components: {
    NotionNestedList: __vue_component__$e,
    NotionTextRenderer: __vue_component__$p
  },
  computed: _objectSpread2(_objectSpread2({}, blockComputed), {}, {
    start: function start() {
      var _this$value;

      return getListNumber((_this$value = this.value) === null || _this$value === void 0 ? void 0 : _this$value.id, this.blockMap);
    },
    isTopLevel: function isTopLevel() {
      var _this$blockMap$this$v, _this$blockMap$this$v2, _this$value2;

      return this.type !== ((_this$blockMap$this$v = this.blockMap[(_this$value2 = this.value) === null || _this$value2 === void 0 ? void 0 : _this$value2.parent_id]) === null || _this$blockMap$this$v === void 0 ? void 0 : (_this$blockMap$this$v2 = _this$blockMap$this$v.value) === null || _this$blockMap$this$v2 === void 0 ? void 0 : _this$blockMap$this$v2.type);
    }
  })
};/* script */
var __vue_script__$d = script$d;
/* template */

var __vue_render__$c = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _vm.isTopLevel && _vm.type === 'bulleted_list' ? _c('ul', {
    staticClass: "notion-list notion-list-disc"
  }, [_vm._ssrNode("<li>", "</li>", [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1), _vm._ssrNode(" "), _vm.value.content ? _c('NotionNestedList', _vm._b({}, 'NotionNestedList', _vm.pass, false), [_vm._t("default")], 2) : _vm._e()], 2) : _vm.isTopLevel && _vm.type === 'numbered_list' ? _c('ol', {
    staticClass: "notion-list notion-list-numbered",
    attrs: {
      "start": _vm.start
    }
  }, [_vm._ssrNode("<li>", "</li>", [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1), _vm._ssrNode(" "), _vm.value.content ? _c('NotionNestedList', _vm._b({}, 'NotionNestedList', _vm.pass, false), [_vm._t("default")], 2) : _vm._e()], 2) : _c('span', [_vm._ssrNode("<li>", "</li>", [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1), _vm._ssrNode(" "), _vm.value.content ? _c('NotionNestedList', _vm._b({}, 'NotionNestedList', _vm.pass, false), [_vm._t("default")], 2) : _vm._e()], 2);
};

var __vue_staticRenderFns__$c = [];
/* style */

var __vue_inject_styles__$d = undefined;
/* scoped */

var __vue_scope_id__$d = undefined;
/* module identifier */

var __vue_module_identifier__$d = "data-v-0dd7aaec";
/* functional template */

var __vue_is_functional_template__$d = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$d = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$c,
  staticRenderFns: __vue_staticRenderFns__$c
}, __vue_inject_styles__$d, __vue_script__$d, __vue_scope_id__$d, __vue_is_functional_template__$d, __vue_module_identifier__$d, false, undefined, undefined, undefined);//
var script$c = {
  extends: Blockable,
  name: "NotionPageHeader",
  components: {
    Decorator: __vue_component__$q
  }
};/* script */
var __vue_script__$c = script$c;
/* template */

var __vue_render__$b = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('header', {
    staticClass: "notion-page-header"
  }, [_vm._ssrNode("<div class=\"notion-nav-breadcrumbs\"></div>")]);
};

var __vue_staticRenderFns__$b = [];
/* style */

var __vue_inject_styles__$c = undefined;
/* scoped */

var __vue_scope_id__$c = undefined;
/* module identifier */

var __vue_module_identifier__$c = "data-v-1bc93e8b";
/* functional template */

var __vue_is_functional_template__$c = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$c = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$b,
  staticRenderFns: __vue_staticRenderFns__$b
}, __vue_inject_styles__$c, __vue_script__$c, __vue_scope_id__$c, __vue_is_functional_template__$c, __vue_module_identifier__$c, false, undefined, undefined, undefined);var script$b = {
  extends: Blockable,
  name: "NotionPage",
  components: {
    NotionPageHeader: __vue_component__$c,
    NotionPageIcon: __vue_component__$n,
    NotionTextRenderer: __vue_component__$p
  },
  computed: _objectSpread2(_objectSpread2({}, blockComputed), {}, {
    coverStyle: function coverStyle() {
      var coverPosition = (1 - (this.format.page_cover_position || 0.5)) * 100;
      return {
        objectPosition: "center ".concat(coverPosition, "%")
      };
    }
  })
};/* script */
var __vue_script__$b = script$b;
/* template */

var __vue_render__$a = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _vm.level === 0 && _vm.fullPage ? _c('div', {
    staticClass: "notion"
  }, [_vm._ssrNode((_vm.format && _vm.format.page_cover ? "<img" + _vm._ssrAttr("alt", _vm.getTextContent(_vm.title)) + _vm._ssrAttr("src", _vm.mapImageUrl(_vm.format.page_cover, _vm.block)) + " class=\"notion-page-cover\"" + _vm._ssrStyle(null, _vm.coverStyle, null) + ">" : "<!---->") + " "), _vm._ssrNode("<main" + _vm._ssrClass(null, ['notion-page', _vm.format && !_vm.format.page_cover && 'notion-page-offset', _vm.format && _vm.format.page_full_width && 'notion-full-width', _vm.format && _vm.format.page_small_text && 'notion-small-text']) + ">", "</main>", [_c('NotionPageIcon', _vm._b({
    attrs: {
      "big": ""
    }
  }, 'NotionPageIcon', _vm.pass, false)), _vm._ssrNode(" "), _vm._ssrNode("<div class=\"notion-title\">", "</div>", [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1), _vm._ssrNode(" "), _vm._t("default")], 2)], 2) : _vm.level === 0 ? _c('main', {
    staticClass: "notion"
  }, [_vm._t("default")], 2) : _vm.hasPageLinkOptions ? _c(_vm.pageLinkOptions.component, _vm._b({
    tag: "component",
    staticClass: "notion-page-link"
  }, 'component', _vm.pageLinkProps(_vm.value.id), false), [_c('div', {
    staticClass: "notion-page-icon"
  }, [_c('NotionPageIcon', _vm._b({}, 'NotionPageIcon', _vm.pass, false))], 1), _vm._v(" "), _c('div', {
    staticClass: "notion-page-text"
  }, [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1)]) : _c('a', {
    staticClass: "notion-page-link",
    attrs: {
      "target": _vm.pageLinkTarget,
      "href": _vm.mapPageUrl(_vm.value.id)
    }
  }, [_vm._ssrNode("<div class=\"notion-page-icon\">", "</div>", [_c('NotionPageIcon', _vm._b({}, 'NotionPageIcon', _vm.pass, false))], 1), _vm._ssrNode(" "), _vm._ssrNode("<div class=\"notion-page-text\">", "</div>", [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1)], 2);
};

var __vue_staticRenderFns__$a = [];
/* style */

var __vue_inject_styles__$b = undefined;
/* scoped */

var __vue_scope_id__$b = undefined;
/* module identifier */

var __vue_module_identifier__$b = "data-v-4849e480";
/* functional template */

var __vue_is_functional_template__$b = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$b = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$a,
  staticRenderFns: __vue_staticRenderFns__$a
}, __vue_inject_styles__$b, __vue_script__$b, __vue_scope_id__$b, __vue_is_functional_template__$b, __vue_module_identifier__$b, false, undefined, undefined, undefined);//
var script$a = {
  extends: Blockable,
  name: "NotionQuote",
  components: {
    NotionTextRenderer: __vue_component__$p
  }
};/* script */
var __vue_script__$a = script$a;
/* template */

var __vue_render__$9 = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _vm.properties ? _c('blockquote', {
    staticClass: "notion-quote"
  }, [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1) : _vm._e();
};

var __vue_staticRenderFns__$9 = [];
/* style */

var __vue_inject_styles__$a = undefined;
/* scoped */

var __vue_scope_id__$a = undefined;
/* module identifier */

var __vue_module_identifier__$a = "data-v-bf6bbaf4";
/* functional template */

var __vue_is_functional_template__$a = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$a = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$9,
  staticRenderFns: __vue_staticRenderFns__$9
}, __vue_inject_styles__$a, __vue_script__$a, __vue_scope_id__$a, __vue_is_functional_template__$a, __vue_module_identifier__$a, false, undefined, undefined, undefined);var script$9 = {
  extends: Blockable,
  name: "NotionSyncPointer",
  computed: _objectSpread2(_objectSpread2({}, blockComputed), {}, {
    referencePointerId: function referencePointerId() {
      var _this$format, _this$format$transclu;

      return (_this$format = this.format) === null || _this$format === void 0 ? void 0 : (_this$format$transclu = _this$format.transclusion_reference_pointer) === null || _this$format$transclu === void 0 ? void 0 : _this$format$transclu.id;
    }
  })
};/* script */
var __vue_script__$9 = script$9;
/* template */

var __vue_render__$8 = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('NotionRenderer', _vm._b({
    attrs: {
      "blockMap": _vm.blockMap,
      "contentId": _vm.referencePointerId
    }
  }, 'NotionRenderer', _vm.pass, false));
};

var __vue_staticRenderFns__$8 = [];
/* style */

var __vue_inject_styles__$9 = undefined;
/* scoped */

var __vue_scope_id__$9 = undefined;
/* module identifier */

var __vue_module_identifier__$9 = "data-v-534f19e5";
/* functional template */

var __vue_is_functional_template__$9 = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$9 = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$8,
  staticRenderFns: __vue_staticRenderFns__$8
}, __vue_inject_styles__$9, __vue_script__$9, __vue_scope_id__$9, __vue_is_functional_template__$9, __vue_module_identifier__$9, false, undefined, undefined, undefined);//
var script$8 = {
  extends: Blockable,
  name: "NotionTable"
};/* script */
var __vue_script__$8 = script$8;
/* template */

var __vue_render__$7 = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('div', {
    staticClass: "notion-simple-table-wrapper"
  }, [_vm._ssrNode("<table class=\"notion-simple-table\">", "</table>", [_vm._ssrNode("<tbody>", "</tbody>", [_vm._t("default")], 2)])]);
};

var __vue_staticRenderFns__$7 = [];
/* style */

var __vue_inject_styles__$8 = undefined;
/* scoped */

var __vue_scope_id__$8 = undefined;
/* module identifier */

var __vue_module_identifier__$8 = "data-v-e70e2b32";
/* functional template */

var __vue_is_functional_template__$8 = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$8 = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$7,
  staticRenderFns: __vue_staticRenderFns__$7
}, __vue_inject_styles__$8, __vue_script__$8, __vue_scope_id__$8, __vue_is_functional_template__$8, __vue_module_identifier__$8, false, undefined, undefined, undefined);var script$7 = {
  extends: Blockable,
  name: "NotionTableRow",
  components: {
    NotionTextRenderer: __vue_component__$p
  },
  computed: _objectSpread2(_objectSpread2({}, blockComputed), {}, {
    hasHeaderColumn: function hasHeaderColumn() {
      var _this$parent, _this$parent$value, _this$parent$value$fo;

      return (_this$parent = this.parent) === null || _this$parent === void 0 ? void 0 : (_this$parent$value = _this$parent.value) === null || _this$parent$value === void 0 ? void 0 : (_this$parent$value$fo = _this$parent$value.format) === null || _this$parent$value$fo === void 0 ? void 0 : _this$parent$value$fo.table_block_column_header;
    },
    hasHeaderRow: function hasHeaderRow() {
      var _this$parent2, _this$parent2$value, _this$parent2$value$f;

      return (_this$parent2 = this.parent) === null || _this$parent2 === void 0 ? void 0 : (_this$parent2$value = _this$parent2.value) === null || _this$parent2$value === void 0 ? void 0 : (_this$parent2$value$f = _this$parent2$value.format) === null || _this$parent2$value$f === void 0 ? void 0 : _this$parent2$value$f.table_block_row_header;
    },
    columns: function columns() {
      var _this$parent3, _this$parent3$value, _this$parent3$value$f;

      return (_this$parent3 = this.parent) === null || _this$parent3 === void 0 ? void 0 : (_this$parent3$value = _this$parent3.value) === null || _this$parent3$value === void 0 ? void 0 : (_this$parent3$value$f = _this$parent3$value.format) === null || _this$parent3$value$f === void 0 ? void 0 : _this$parent3$value$f.table_block_column_order;
    }
  }),
  methods: {
    cell: function cell(columnId) {
      var _this$properties$colu, _this$properties;

      // return empty notion decorated text if row is empty
      return (_this$properties$colu = this === null || this === void 0 ? void 0 : (_this$properties = this.properties) === null || _this$properties === void 0 ? void 0 : _this$properties[columnId]) !== null && _this$properties$colu !== void 0 ? _this$properties$colu : [[" ", false]];
    },
    isHeader: function isHeader(columnIndex) {
      return this.hasHeaderColumn && this.contentIndex == 0 || this.hasHeaderRow && columnIndex == 0;
    }
  }
};/* script */
var __vue_script__$7 = script$7;
/* template */

var __vue_render__$6 = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('tr', {
    staticClass: "notion-simple-table-row"
  }, _vm._l(_vm.columns, function (columnId, columnIndex) {
    return _vm._ssrNode("<td class=\"notion-simple-table-data\">", "</td>", [_vm._ssrNode("<div" + _vm._ssrClass(null, {
      'notion-simple-table-header': _vm.isHeader(columnIndex)
    }) + ">", "</div>", [_vm._ssrNode("<div class=\"notion-simple-table-cell-text\">", "</div>", [_c('NotionTextRenderer', _vm._b({
      attrs: {
        "text": _vm.cell(columnId)
      }
    }, 'NotionTextRenderer', _vm.pass, false))], 1)])]);
  }), 0);
};

var __vue_staticRenderFns__$6 = [];
/* style */

var __vue_inject_styles__$7 = undefined;
/* scoped */

var __vue_scope_id__$7 = undefined;
/* module identifier */

var __vue_module_identifier__$7 = "data-v-f7b08f8e";
/* functional template */

var __vue_is_functional_template__$7 = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$7 = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$6,
  staticRenderFns: __vue_staticRenderFns__$6
}, __vue_inject_styles__$7, __vue_script__$7, __vue_scope_id__$7, __vue_is_functional_template__$7, __vue_module_identifier__$7, false, undefined, undefined, undefined);//
var script$6 = {
  extends: Blockable,
  name: "NotionText",
  components: {
    NotionTextRenderer: __vue_component__$p
  }
};/* script */
var __vue_script__$6 = script$6;
/* template */

var __vue_render__$5 = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _vm.properties ? _c('p', {
    class: ['notion-text', _vm.blockColorClass()]
  }, [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1) : _c('div', {
    staticClass: "notion-blank"
  }, [_vm._ssrNode(" ")]);
};

var __vue_staticRenderFns__$5 = [];
/* style */

var __vue_inject_styles__$6 = undefined;
/* scoped */

var __vue_scope_id__$6 = undefined;
/* module identifier */

var __vue_module_identifier__$6 = "data-v-219aa3c0";
/* functional template */

var __vue_is_functional_template__$6 = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$6 = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$5,
  staticRenderFns: __vue_staticRenderFns__$5
}, __vue_inject_styles__$6, __vue_script__$6, __vue_scope_id__$6, __vue_is_functional_template__$6, __vue_module_identifier__$6, false, undefined, undefined, undefined);//
var script$5 = {
  extends: Blockable,
  name: "NotionTodo",
  components: {
    NotionTextRenderer: __vue_component__$p
  }
};/* script */
var __vue_script__$5 = script$5;
/* template */

var __vue_render__$4 = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('div', [_vm._ssrNode("<input type=\"checkbox\" disabled=\"disabled\"" + _vm._ssrAttr("value", _vm.title) + _vm._ssrAttr("checked", _vm.properties.checked) + "> "), _vm._ssrNode("<label>", "</label>", [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1)], 2);
};

var __vue_staticRenderFns__$4 = [];
/* style */

var __vue_inject_styles__$5 = undefined;
/* scoped */

var __vue_scope_id__$5 = undefined;
/* module identifier */

var __vue_module_identifier__$5 = "data-v-0abd2154";
/* functional template */

var __vue_is_functional_template__$5 = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$5 = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$4,
  staticRenderFns: __vue_staticRenderFns__$4
}, __vue_inject_styles__$5, __vue_script__$5, __vue_scope_id__$5, __vue_is_functional_template__$5, __vue_module_identifier__$5, false, undefined, undefined, undefined);//
var script$4 = {
  extends: Blockable,
  name: "NotionToggle",
  components: {
    NotionTextRenderer: __vue_component__$p
  }
};/* script */
var __vue_script__$4 = script$4;
/* template */

var __vue_render__$3 = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('details', {
    staticClass: "notion-toggle"
  }, [_vm._ssrNode("<summary>", "</summary>", [_c('NotionTextRenderer', _vm._b({
    attrs: {
      "text": _vm.title
    }
  }, 'NotionTextRenderer', _vm.pass, false))], 1), _vm._ssrNode(" "), _vm._ssrNode("<div>", "</div>", [_vm._t("default")], 2)], 2);
};

var __vue_staticRenderFns__$3 = [];
/* style */

var __vue_inject_styles__$4 = undefined;
/* scoped */

var __vue_scope_id__$4 = undefined;
/* module identifier */

var __vue_module_identifier__$4 = "data-v-1c31b83d";
/* functional template */

var __vue_is_functional_template__$4 = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$4 = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$3,
  staticRenderFns: __vue_staticRenderFns__$3
}, __vue_inject_styles__$4, __vue_script__$4, __vue_scope_id__$4, __vue_is_functional_template__$4, __vue_module_identifier__$4, false, undefined, undefined, undefined);var _components;
var script$3 = {
  extends: Blockable,
  name: "NotionBlock",
  components: (_components = {
    NotionBookmark: __vue_component__$o,
    NotionCallout: __vue_component__$m,
    NotionCode: __vue_component__$l,
    NotionColumn: __vue_component__$k,
    NotionEquation: __vue_component__$j
  }, _defineProperty(_components, "NotionEquation", __vue_component__$j), _defineProperty(_components, "NotionFigure", __vue_component__$g), _defineProperty(_components, "NotionHeader", __vue_component__$f), _defineProperty(_components, "NotionList", __vue_component__$d), _defineProperty(_components, "NotionPage", __vue_component__$b), _defineProperty(_components, "NotionQuote", __vue_component__$a), _defineProperty(_components, "NotionSyncPointer", __vue_component__$9), _defineProperty(_components, "NotionTable", __vue_component__$8), _defineProperty(_components, "NotionTableRow", __vue_component__$7), _defineProperty(_components, "NotionText", __vue_component__$6), _defineProperty(_components, "NotionTodo", __vue_component__$5), _defineProperty(_components, "NotionToggle", __vue_component__$4), _components),
  computed: _objectSpread2(_objectSpread2({}, blockComputed), {}, {
    isRendererRegistered: function isRendererRegistered() {
      var _Vue$options;

      return "NotionRenderer" in (Vue__default['default'] === null || Vue__default['default'] === void 0 ? void 0 : (_Vue$options = Vue__default['default'].options) === null || _Vue$options === void 0 ? void 0 : _Vue$options.components);
    }
  })
};/* script */
var __vue_script__$3 = script$3;
/* template */

var __vue_render__$2 = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _vm.blockOverrides.hasOwnProperty(_vm.type) ? _c(_vm.blockOverrides[_vm.type], _vm._b({
    tag: "component"
  }, 'component', _vm.pass, false)) : _vm.isType('page') ? _c('div', [_c('NotionPage', _vm._b({}, 'NotionPage', _vm.pass, false), [_vm._t("default")], 2)], 1) : _vm.isType(['header', 'sub_header', 'sub_sub_header']) ? _c('NotionHeader', _vm._b({}, 'NotionHeader', _vm.pass, false)) : _vm.isType('bookmark') ? _c('NotionBookmark', _vm._b({}, 'NotionBookmark', _vm.pass, false)) : _vm.isType('callout') ? _c('NotionCallout', _vm._b({}, 'NotionCallout', _vm.pass, false)) : _vm.isType('code') ? _c('NotionCode', _vm._b({}, 'NotionCode', _vm.pass, false)) : _vm.isType('equation') ? _c('NotionEquation', _vm._b({}, 'NotionEquation', _vm.pass, false)) : _vm.isType('text') ? _c('NotionText', _vm._b({}, 'NotionText', _vm.pass, false)) : _vm.isType('quote') ? _c('NotionQuote', _vm._b({}, 'NotionQuote', _vm.pass, false)) : _vm.isType('to_do') ? _c('NotionTodo', _vm._b({}, 'NotionTodo', _vm.pass, false)) : _vm.isType('toggle') ? _c('NotionToggle', _vm._b({}, 'NotionToggle', _vm.pass, false), [_vm._t("default")], 2) : _vm.isType('column_list') ? _c('div', {
    staticClass: "notion-row"
  }, [_vm._t("default")], 2) : _vm.isType('column') ? _c('NotionColumn', {
    attrs: {
      "format": _vm.format
    }
  }, [_vm._t("default")], 2) : _vm.isType(['bulleted_list', 'numbered_list']) ? _c('NotionList', _vm._b({}, 'NotionList', _vm.pass, false), [_vm._t("default")], 2) : _vm.isType(['image', 'embed', 'figma', 'video', 'audio']) ? _c('NotionFigure', _vm._b({}, 'NotionFigure', _vm.pass, false)) : _vm.isType('table') ? _c('NotionTable', _vm._b({}, 'NotionTable', _vm.pass, false), [_vm._t("default")], 2) : _vm.isRendererRegistered && _vm.isType('transclusion_reference') ? _c('NotionSyncPointer', _vm._b({}, 'NotionSyncPointer', _vm.pass, false)) : _vm.isRendererRegistered && _vm.isType('transclusion_container') ? _c('div', {
    staticClass: "notion-sync-block"
  }, [_vm._t("default")], 2) : _vm.isType('table_row') ? _c('NotionTableRow', _vm._b({}, 'NotionTableRow', _vm.pass, false)) : _vm.isType('divider') ? _c('hr', {
    staticClass: "notion-hr"
  }) : _vm.todo && _vm.visible ? _c('div', [_vm._v("\n  todo: " + _vm._s(_vm.type) + "\n  "), _vm._t("default")], 2) : _vm._e();
};

var __vue_staticRenderFns__$2 = [];
/* style */

var __vue_inject_styles__$3 = undefined;
/* scoped */

var __vue_scope_id__$3 = undefined;
/* module identifier */

var __vue_module_identifier__$3 = "data-v-b746bdc4";
/* functional template */

var __vue_is_functional_template__$3 = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$3 = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$2,
  staticRenderFns: __vue_staticRenderFns__$2
}, __vue_inject_styles__$3, __vue_script__$3, __vue_scope_id__$3, __vue_is_functional_template__$3, __vue_module_identifier__$3, false, undefined, undefined, undefined);//
var script$2 = {
  extends: Blockable,
  name: "NotionRenderer",
  components: {
    NotionBlock: __vue_component__$3
  },
  props: {
    blockMap: [Object],
    contentId: String,
    contentIndex: {
      type: Number,
      default: 0
    },
    fullPage: {
      type: Boolean,
      default: false
    },
    hideList: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    level: {
      type: Number,
      default: 0
    },
    mapImageUrl: {
      type: Function,
      default: defaultMapImageUrl
    },
    mapPageUrl: {
      type: Function,
      default: defaultMapPageUrl
    },
    pageLinkOptions: Object,
    imageOptions: Object,
    prism: {
      type: Boolean,
      default: false
    },
    todo: {
      type: Boolean,
      default: false
    }
  }
};/* script */
var __vue_script__$2 = script$2;
/* template */

var __vue_render__$1 = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _vm.blockMap && _vm.value ? _c('NotionBlock', _vm._b({}, 'NotionBlock', _vm.pass, false), _vm._l(_vm.value.content, function (contentId, contentIndex) {
    return _c('NotionRenderer', _vm._b({
      key: contentId,
      attrs: {
        "level": _vm.level + 1,
        "content-id": contentId,
        "content-index": contentIndex
      }
    }, 'NotionRenderer', _vm.pass, false));
  }), 1) : _vm._e();
};

var __vue_staticRenderFns__$1 = [];
/* style */

var __vue_inject_styles__$2 = undefined;
/* scoped */

var __vue_scope_id__$2 = undefined;
/* module identifier */

var __vue_module_identifier__$2 = "data-v-06df2879";
/* functional template */

var __vue_is_functional_template__$2 = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$2 = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$1,
  staticRenderFns: __vue_staticRenderFns__$1
}, __vue_inject_styles__$2, __vue_script__$2, __vue_scope_id__$2, __vue_is_functional_template__$2, __vue_module_identifier__$2, false, undefined, undefined, undefined);//
//
//
//
var baseWidth = 46; // todo: add magic numbers to a config json

var script$1 = {
  name: "NotionColumnSpacer",
  computed: {
    spacerStyle: function spacerStyle() {
      return {
        width: "".concat(baseWidth, "px")
      };
    }
  }
};/* script */
var __vue_script__$1 = script$1;
/* template */

var __vue_render__ = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('div', {
    staticClass: "notion-spacer",
    style: _vm.spacerStyle
  }, []);
};

var __vue_staticRenderFns__ = [];
/* style */

var __vue_inject_styles__$1 = undefined;
/* scoped */

var __vue_scope_id__$1 = undefined;
/* module identifier */

var __vue_module_identifier__$1 = "data-v-36a5f7d0";
/* functional template */

var __vue_is_functional_template__$1 = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__$1 = /*#__PURE__*/normalizeComponent({
  render: __vue_render__,
  staticRenderFns: __vue_staticRenderFns__
}, __vue_inject_styles__$1, __vue_script__$1, __vue_scope_id__$1, __vue_is_functional_template__$1, __vue_module_identifier__$1, false, undefined, undefined, undefined);var script = {
  functional: true,
  render: function render(h, ctx) {
    return ctx.children;
  }
};/* script */
var __vue_script__ = script;
/* template */

/* style */

var __vue_inject_styles__ = undefined;
/* scoped */

var __vue_scope_id__ = undefined;
/* module identifier */

var __vue_module_identifier__ = "data-v-417b8fe7";
/* functional template */

var __vue_is_functional_template__ = undefined;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

var __vue_component__ = /*#__PURE__*/normalizeComponent({}, __vue_inject_styles__, __vue_script__, __vue_scope_id__, __vue_is_functional_template__, __vue_module_identifier__, false, undefined, undefined, undefined);/* eslint-disable import/prefer-default-export */var components=/*#__PURE__*/Object.freeze({__proto__:null,NotionRenderer: __vue_component__$2,NotionBookmark: __vue_component__$o,NotionCallout: __vue_component__$m,NotionCode: __vue_component__$l,NotionColumn: __vue_component__$k,NotionDecorator: __vue_component__$q,NotionEquation: __vue_component__$j,NotionHeader: __vue_component__$f,NotionList: __vue_component__$d,NotionPage: __vue_component__$b,NotionQuote: __vue_component__$a,NotionTable: __vue_component__$8,NotionTableRow: __vue_component__$7,NotionText: __vue_component__$6,NotionTodo: __vue_component__$5,NotionToggle: __vue_component__$4,NotionAsset: __vue_component__$i,NotionColumnSpacer: __vue_component__$1,NotionFigure: __vue_component__$g,NotionFragment: __vue_component__,NotionImage: __vue_component__$h,NotionNestedList: __vue_component__$e,NotionPageHeader: __vue_component__$c,NotionPageIcon: __vue_component__$n,NotionTextRenderer: __vue_component__$p});var getPageTable = /*#__PURE__*/function () {
  var _ref = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(pageId) {
    var apiUrl,
        _args = arguments;
    return regeneratorRuntime.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            apiUrl = _args.length > 1 && _args[1] !== undefined ? _args[1] : "https://api.vue-notion.workers.dev/v1";
            _context.next = 3;
            return fetch__default['default']("".concat(apiUrl, "/table/").concat(pageId)).then(function (res) {
              return res.json();
            });

          case 3:
            return _context.abrupt("return", _context.sent);

          case 4:
          case "end":
            return _context.stop();
        }
      }
    }, _callee);
  }));

  return function getPageTable(_x) {
    return _ref.apply(this, arguments);
  };
}();

var getPageBlocks = /*#__PURE__*/function () {
  var _ref2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee2(pageId) {
    var apiUrl,
        _args2 = arguments;
    return regeneratorRuntime.wrap(function _callee2$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            apiUrl = _args2.length > 1 && _args2[1] !== undefined ? _args2[1] : "https://api.vue-notion.workers.dev/v1";
            _context2.next = 3;
            return fetch__default['default']("".concat(apiUrl, "/page/").concat(pageId)).then(function (res) {
              return res.json();
            });

          case 3:
            return _context2.abrupt("return", _context2.sent);

          case 4:
          case "end":
            return _context2.stop();
        }
      }
    }, _callee2);
  }));

  return function getPageBlocks(_x2) {
    return _ref2.apply(this, arguments);
  };
}();var install = function installVueNotion(Vue) {
  if (install.installed) return;
  install.installed = true;
  Object.entries(components).forEach(function (_ref) {
    var _ref2 = _slicedToArray(_ref, 2),
        componentName = _ref2[0],
        component = _ref2[1];

    Vue.component(componentName, component);
  });
}; // Create module definition for Vue.use()


var plugin = {
  install: install
}; // To auto-install on non-es builds, when vue is found
// eslint-disable-next-line no-redeclare

/* global window, global */

{
  var GlobalVue = null;

  if (typeof window !== "undefined") {
    GlobalVue = window.Vue;
  } else if (typeof global !== "undefined") {
    GlobalVue = global.Vue;
  }

  if (GlobalVue) {
    GlobalVue.use(plugin);
  }
} // Default export is library as a whole, registered via Vue.use()
 // todo: remove .esm from default packaging for cleaner imports
exports.Blockable=Blockable;exports.NotionAsset=__vue_component__$i;exports.NotionBookmark=__vue_component__$o;exports.NotionCallout=__vue_component__$m;exports.NotionCode=__vue_component__$l;exports.NotionColumn=__vue_component__$k;exports.NotionColumnSpacer=__vue_component__$1;exports.NotionDecorator=__vue_component__$q;exports.NotionEquation=__vue_component__$j;exports.NotionFigure=__vue_component__$g;exports.NotionFragment=__vue_component__;exports.NotionHeader=__vue_component__$f;exports.NotionImage=__vue_component__$h;exports.NotionList=__vue_component__$d;exports.NotionNestedList=__vue_component__$e;exports.NotionPage=__vue_component__$b;exports.NotionPageHeader=__vue_component__$c;exports.NotionPageIcon=__vue_component__$n;exports.NotionQuote=__vue_component__$a;exports.NotionRenderer=__vue_component__$2;exports.NotionTable=__vue_component__$8;exports.NotionTableRow=__vue_component__$7;exports.NotionText=__vue_component__$6;exports.NotionTextRenderer=__vue_component__$p;exports.NotionTodo=__vue_component__$5;exports.NotionToggle=__vue_component__$4;exports.blockComputed=blockComputed;exports.blockProps=blockProps;exports['default']=plugin;exports.defaultMapImageUrl=defaultMapImageUrl;exports.defaultMapPageUrl=defaultMapPageUrl;exports.getListNumber=getListNumber;exports.getPageBlocks=getPageBlocks;exports.getPageTable=getPageTable;exports.getTextContent=getTextContent;
